File: src/test/java/work/lclpnet/ap2/game/fine_tuning/melody/ScaleTest.java
package work.lclpnet.ap2.game.fine_tuning.melody;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static work.lclpnet.ap2.game.fine_tuning.melody.Note.*;

class ScaleTest {

    @Test
    void cMajor() {
        var notes = Scale.getMajorNotes(C4);

        assertArrayEquals(new Note[] {C4, D4, E4, F4, G4, A4, B4, C5}, notes);
    }
}

File: src/minigames/fine_tuning/src/test/kotlin/work/lclpnet/ap2/game/fine_tuning/melody/ScaleTest.kt
package work.lclpnet.ap2.game.fine_tuning.melody

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ScaleTest {

    @Test
    fun cMajor() {
        val notes = Scale.getMajorNotes(Note.C4)
        assertEquals(listOf(Note.C4, Note.D4, Note.E4, Note.F4, Note.G4, Note.A4, Note.B4, Note.C5), notes)
    }
}
